CMSgov/cmcs-eregulations
# EREGCSC-1617 -- Regulations Search Enabler

Resolves [EREGCSC-1617](https://jiraent.cms.gov/browse/EREGCSC-1617)

**Description**

Creates new `search` django template and uses Vite/Vue to recreate existing Regulations Search functionality in Single Page App paradigm that matches existing resources page.

**This pull request changes:**

- Adds `search_new.html` django template as foundation for new combined search page
- Adds logic to `search.py` view to reshape search results and synonym results from Django QuerySets into lists of object or strings that can be easily output to JSON to be ingested by JS/Vue
- Creates new `Search.vue` component and associated `/search` route for new Search SPA
- Recreates existing Search functionality in new Search SPA
- Follows styles defined in Combined Search Figma comps

**Steps to manually verify this change**

1. Visit `/search/` on this experimental deployment and make sure it works as expected
